src/backend/src/game/game.controller.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Post,
+  Sse,
+  MessageEvent,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Player } from '../entities/player.entity';
+import { Repository } from 'typeorm';
+import { Game } from '../entities/game.entity';
+import { Grab } from '../entities/grab.entity';
+import { Round } from '../entities/round.entity';
+import { interval, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Controller('game')
+export class GameController {
+  constructor(
+    @InjectRepository(Player)
+    private playersRepository: Repository<Player>,
+    @InjectRepository(Round)
+    private roundsRepository: Repository<Round>,
+  ) {}
+
+  @Post('take')
+  async take(
+    @Body('playerId') playerId: number,
+    @Body('howMany') howMany: number,
+    @Body('timeTaken') timeTaken: number,
+    @Body('roundId') roundId: number,
+  ): Promise<number> {
+    const player = await this.playersRepository.findOne(playerId);
+    if (!player) {
+      throw new BadRequestException('Player does not exist');
+    }
+
+    const round = await this.roundsRepository.findOne(roundId);
+    if (!round) {
+      throw new BadRequestException('Round does not exist');
+    }
+
+    const grab = await Grab.create();
+
+    grab.round = round;
+    grab.player = player;
+    grab.howMany = howMany;
+    grab.timeTaken = timeTaken;
```

Comment:
I think you can add these params to grab.create ie.
```
Grab.create({round, player, howMany, timeTaken})
```

Reply:
ok this was causing webstorm to throw a tantrum yesterday, but it works now so that's cool